### ISSUE TITLE
bin/maxmind-geolite-mirror making many queries, ending up in error 429

### ISSUE THREAD

**Issue by DoobleD** [msimerson/maxmind-geolite-mirror#47]
Running bin/maxmind-geolite-mirror **once** gives the following error:

```
MAXMIND_LICENSE_KEY=>obfusacated>  ./bin/maxmind-geolite-mirror 
downloading /usr/local/share/GeoIP/GeoLite2-Country.mmdb
HEADERS: {"date":"Sat, 22 Jun 2024 20:56:34 GMT","content-type":"text/plain;charset=UTF-8","content-length":"44","connection":"close","cache-control":"private, max-age=0","expires":"Sat, 22 Jun 2024 20:56:34 GMT","strict-transport-security":"max-age=31536000; includeSubDomains; preload","server":"cloudflare","cf-ray":"897f2551db459ec2-CDG"}
response code 429 not handled!
exiting prematurely.
```

This has been tested with a fresh clone of the repo, and a fresh Maxmind GeoLite2 account with a new license key. Following the error we can see on the Maxmind account that 65 queries were made, which exceeds the 30 queries per day limit for GeoLite accounts.

![image](https://github.com/msimerson/maxmind-geolite-mirror/assets/918201/7ec11464-5292-47b5-9de5-0401f12bca8c)

Maxmind also sends the following email:

![image](https://github.com/msimerson/maxmind-geolite-mirror/assets/918201/e31b8c35-3ca1-4fbc-bd47-3a47df46ae4c)

**Comment by msimerson** [comment]
I'm aware, I just haven't dug deep enough to figure out exactly why.

**Comment by msimerson** [comment]
Also, a workaround if you're using this with MT6 is to set `GEOIP_UPDATER=geoipupdate` in mail-toaster.conf, and use the maxmind client.

**Comment by keithws** [comment]
I am also receiving this issue.

In my situation, the script (run at 6am by cron) is able to make 289 requests within about 20 seconds for the exact same file (the country file) over and over again.

**Comment by msimerson** [comment]
Yes, the retry it's because Maxmind changed how they redirect clients after this was written. Some time needs to be spent handling them better.
